feat(gtfs): add GtfsService for scheduled GO Transit GTFS download/extraction to memory

adinschmidt opened the pull request:
# Pull Request

## 📋 Description

- Downloads `GO-GTFS.zip` from Metrolinx CDN
- Extracts text files in-memory
- Runs on module init (startup) + every 6 hours
- Exposes `getLatestGtfsFiles()` for downstream parsing
## 🔗 Related Issue

https://github.com/CivicTechWR/go-train-group-pass/issues/45

## 🧪 Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Accessibility improvement

jliu1016 commented on backend/src/gtfs/gtfs.service.spec.ts:
```diff
@@ -0,0 +1,188 @@
+/* eslint-disable @typescript-eslint/unbound-method, @typescript-eslint/no-unsafe-assignment */
```
what's the reason for the eslint disable here?

jliu1016 commented:
I think the process flow for this is

On trigger (currently being debated but I think it should be whenever someone makes any request related to schedule that it does a quick check of if the schedule data is outdated) -> Download txt files for GTFS -> parse each txt file into the corresponding entity 

Also I noticed the serviceid is the calendar date... should we add an internal UUID to the calendar_date entity in case of conflicts? I'm not sure if we want to be overwriting rows of existing schedules and if we are we probably need a logger system in place.

adinschmidt commented:
I have addressed the review comments with the following changes:

1.  **ConfigService**: Refactored `GtfsService` to use `ConfigService` for retrieving the `GTFS_URL`.
2.  **Linting**: Fixed the `eslint-disable` in `gtfs.service.spec.ts` by properly typing the `axios` and `jszip` mocks, removing the need for the broad disable.
3.  **Cron Schedule**: Kept the cron schedule as a constant for now. Making it dynamic via `ConfigService` would require using `SchedulerRegistry` which adds complexity. If strictly needed, we can add it later.
4.  **Service ID / Entities**: Regarding the `serviceid` and entity structure, this PR focuses on the download/extraction mechanism. The parsing logic and entity mapping (where we can introduce UUIDs if needed) should be handled in the downstream implementation.
